src/Perlang.ConsoleApp/Program.cs

```diff
@@ -197,11 +200,49 @@ public static int MainWithCustomConsole(string[] args, IConsole console)
                 })
             };
 
-            rootCommand.AddArgument(new Argument
+            var scriptNameArgument = new Argument<string>
             {
-                Arity = ArgumentArity.ZeroOrMore
+                Name = "script-name",
+                Arity = ArgumentArity.ZeroOrOne,
+            };
+
+            scriptNameArgument.AddValidator(result =>
+            {
+                var tokens = result.Parent!.Tokens;
+
+                if (tokens.Any(t => t.Type == System.CommandLine.Parsing.TokenType.Option && t.Value == evalOption.Name))
+                {
+                    return "<script-name> positional argument cannot be used together with the -e option";
+                }
+
+                return null;
             });
 
+            rootCommand.AddArgument(scriptNameArgument);
+
+            rootCommand.AddValidator(result =>
+            {
+                if (result.HasOption(evalOption) && result.HasOption(printOption))
+                {
+                    return "Error: the -e and -p option are mutually exclusive";
+                }
+
+                if (result.HasOption(evalOption) && result.HasArgument(scriptNameArgument))
+                {
+                    return "Error: the -e option cannot be combined with the <script-name> argument";
+                }
+
+                return null;
+            });
+
+            var scriptArguments = new Argument<string>
+            {
+                Name = "args",
+                Arity = ArgumentArity.ZeroOrMore
+            };
+
+            rootCommand.AddArgument(scriptArguments);
+
```

Comment:
Hmm... If we're really at it, these hunks could actually go into yet another separate PR, since it's just "cleanup of commandline parsing-related code" with no semantic changes. (stumbled into this when trying to add the `-Wno-error` option, which is not yet included in this MR.)

Reply:
Said and done: #191